src/main.c

```diff
@@ -21,10 +21,23 @@ void show_error(void)
 	gtk_widget_destroy(dialog);
 }
 
+static void
+activate (GtkApplication *app,
+          gpointer        user_data)
+{
+	GtkWidget *window;
+
+	window = app_win_create(app);
+	gtk_widget_show_all(window);
+	gtk_window_present (GTK_WINDOW (window));
+}
```

Comment:
Please don't mix the coding styles. You are apparently used to a different one, but I don't like the spaces between function name and parenthesis: `gtk_window_present (GTK_WINDOW (window));` and don't do the multiline function headers, unless the line gets too long. I know the GNOME apps look like this -- I find them harder to read. In any case -- the mix of styles is the worst option.